### ISSUE TITLE
Add support for associating a Lambda Function to a cloudfront instance

### ISSUE THREAD

**Issue by wslack** [18F/cf-cdn-service-broker#101]
Per https://github.com/18F/federalist/issues/1485 and specifically https://github.com/18F/federalist/issues/1485#issuecomment-360311922

Federalist is interested in replacing https://github.com/18F/federalist-proxy with a Lambda function associated to its cloudfront instances. However, as seen in the linked discussion, we would like to avoid having to ask the cloud.gov team to manually associate the lambda function to each new service instance we create.

Therefore we're wondering if the Cloud.gov team can create the Lambda function as part of the base Cloud.gov deployment and load it into Lambda in their AWS account.

The Federalist team could then contribute a cdn-broker feature addition that supported a flag which triggers that (already-created) function to be assigned to the CF distribution at distribution-create-time.

**Comment by amaline** [comment]
As a Cloud.Gov user, the Lambda function should be discoverable in some way.  Maybe when the CDN service is queried the last operation could return a message about which lambda function was attached.

**Comment by wslack** [comment]
@amaline agreed; we'd want to be able to trigger the flag. I'm ok if there's no function to unassociate for the MVP though.

**Comment by sharms** [comment]
Hi Will and Al,

This functionality does not currently exist in the service broker today, and I agree that for certain specific use cases it would be valuable.  I have created a JIRA ticket in our backlog (CG-426) which will be reviewed and prioritized by our product team during the next sprint.

We are also open to patches to the service broker, although we will need to run it by compliance prior to merging to master to determine if this change is a significant change or not based on the scope of the patch.

Thanks,

Steve

**Comment by wslack** [comment]
Thank you! cc @jseppi

**Comment by amirbey** [comment]
hi @sharms 
I am working on implementing Lambda as a replacement for Federalist's webrpoxy for which this issue is a dependency.  Can you provide an update on the status of this issue (JIRA ticket CG-436)?
